test/comparisons.jl

```diff
@@ -280,7 +274,7 @@ end
                 @test setdiff(IntervalSet(later), IntervalSet(earlier)) == IntervalSet(expected_xor[2:2])
 
                 # TODO: Sometimes expected_xor would get mutated in this call
-                @test symdiff([earlier], [later]) == expected_xor != union(expected_xor)
+                @test symdiff([earlier], [later]) == expected_xor == union(expected_xor)
```

Comment:
This minor change is generating some major internal debate. In looking over this series of tests again the intent is to show that the `IntervalSet` behaviour works as expected and that the vector of `Interval` may or may not match. It appears that a commit you added (e122bfe) introduced the additional check to see what the resulting value actually is which makes the intent less clear.

In looking over that commit is seems you just cared about having a check to see what the value actually is so by following what I original intended and your intent we should just do:
```suggestion
                @test symdiff([earlier], [later]) == union(expected_xor)
```

Reply:
> resulting value actually is which makes the intent less clear.

Why is it less clear? The issue with the original check is that we were testing with multiple overloaded methods. If you were to read these tests without context you might be confused about why they don't match because our implementation of `union` for vectors of intervals is a bit weird. It seemed prudent to add a more explicit check for when we deprecated the `union` method. 

I suppose they could have been separate checks, but I like that the change is all on one line. Anyway, I agree that going forward we don't need to check both anymore, since `union` works as expected now.

Reply:
When we added the `IntervalSet` set methods tests we also added tests against vectors of intervals just to document any deviations in the behaviour but ultimately we didn't care what the result was. 

This originally looked like:
```julia
@test symdiff([earlier], [later]) != union(expected_xor)
@test symdiff(IntervalSet(earlier), IntervalSet(later)) == union(IntervalSet(expected_xor))
```
The vector of intervals test is just mirroring the `IntervalSet` test below but adjusts the comparison to the current behaviour

> Anyway, I agree that going forward we don't need to check both anymore, since union works as expected now.

Sounds good.